packages/search/provider.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useMemo, useState, createContext, useContext } from 'react'
+import algoliaSearch, { SearchClient } from 'algoliasearch'
+import searchInsights from 'search-insights'
+import { AlgoliaConfigObject } from './types'
+
+const SearchContext = createContext(null)
```

Comment:
Converted `provider` to TypeScript. TypeScript warned about `createContext` needing a default value, so I've used `null`.